### PR TITLE
Adds hydration pack.

### PR DESCRIPTION
**Pull request by Rivet-the-Zombie**
Carries eight units of fluid and four volume of inventory, worn on the
back; non-craftable, found in the same places as other canteens.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Ooh, neat. owo

**Comment by Zireael07** [comment]
Woo!

File: data/json/items/containers.json [hunk 1]
```diff
@@ -216,6 +216,36 @@
...
+        "contains": 8,
+        "armor_data" : {
+        "covers" : ["TORSO"],
+        "coverage" : 15,
+        "material_thickness" : 1,
+        "encumbrance" : 0,
+        "environmental_protection" : 0,
+        "warmth" : 0,
+        "storage" : 4
+        },
```
Comment (Coolthulhu) [inline data/json/items/containers.json]: Uh, did you test it?
This line looks very wrong

**Comment by chaosvolt** [comment]
I'm assuming it's the storage, or is it the material.

Rivet appears to have modeled it off of small backpacks with hydration bladders integrated into them, not merely the hydration bladder by itself.

Though the fact it's RIGID might be the issue here.

File: data/json/items/containers.json (hunk 1 above)
Comment (BevapDin) [inline data/json/items/containers.json]: > This line looks very wrong

Why? It looks correct, it closes the object opened by `"armor_data" : {` at line 234. the content of that object might need indentation.

Comment (Rivet-the-Zombie) [inline data/json/items/containers.json]: It just looks weird because the ````"armor_data"```` is in the middle of the other container entries.

**Comment by Rivet-the-Zombie** [comment]
>I'm assuming it's the storage, or is it the material.

>Rivet appears to have modeled it off of small backpacks with hydration bladders integrated into them, not merely the hydration bladder by itself.

>Though the fact it's RIGID might be the issue here.

>EDIT: And I'm guessing VARSIZE also goes against the conventions for backpack-style items.


All the entries for this item are intentional. It even weighs the same as a RL Camelbak.

File: data/json/items/containers.json (hunk 1 above)
Comment (Coolthulhu) [inline data/json/items/containers.json]: Oh, so it's just the indentation.
Why do we have indentation like that for canteens? I just checked and it looks like water containers have the confusing indentation, while bows have the one we have everywhere else (ie. the block is in an extra tab).

**Comment by chaosvolt** [comment]
Ah derp, I see what happened here. A certain star-spawn is a stickler for standard spacing? o3o

**Comment by kevingranade** [comment]
As is a certain... uh... me.
(Please match surrounding indentation)

**Comment by chaosvolt** [comment]
Hard to make an alliterate joke using Kevin's username, sadly.

In any case, I'm still leery of the rigid flag, but whether the flag makes sense likely depends a lot on the model of hydration pack it's based off of.

**Comment by Rivet-the-Zombie** [comment]
>Please match surrounding indentation

It uses the same indentation as the canteens and waterskins in the same file.

**Comment by chaosvolt** [comment]
Unfortunately, it happens sometimes. Left-overs from an ancient era when current indentation standards weren't wide, people PRing while drunk, the list of potential explanations goes on and on. XP

**Comment by Rivet-the-Zombie** [comment]
>people PRing while drunk

This happens far more often than most of you realize.

**Comment by DavidKeaton** [comment]
*belches* uhhh not me.

**Comment by chaosvolt** [comment]
Bad fox, Dwarf Fortress is a far more wholesome activity for when you're drunk.

**Comment by DavidKeaton** [comment]
T^T
also truth

**Pull request merged**